### DISCUSSION TITLE
How can I catch Exception which would raise in conftest.py

### DISCUSSION THREAD

**Discussion by megachweng** [pytest-dev/pytest#11229] [Q&A]
I've developed a Pytest plugin, which is used to make a http request to report the status of the tests to an external system during execution.   

However, as I have no control over my colleagues' code, they might make some errors in `conftest.py`, which could crash before this plugin is even loaded. Therefore, the external system is unable to receive the test status and stuck there.

```python
@pytest.hookimpl(hookwrapper=True, trylast=True)
def pytest_sessionfinish(session, exitstatus):
    yield
    make_test_end_report_request(xxxxx)

@pytest.hookimpl(tryfirst=True)
def pytest_sessionstart(session):
   make_test_start_report_request(xxxxx)
    
```
In this situation, how can I ensure that I can catch all exceptions and send `error occured` report to the external system?

I also tries using `subprocess`
```python
popen = subprocess.Popen(
        ['pytest'] + params or [],
        stdout=subprocess.PIPE,
        stderr=subprocess.STDOUT,
        universal_newlines=True,
        shell=True
    )

return_code = popen.wait()
if return_code > 0:
    make_http_report(reason='Error occured')


```
but still can not 'try catch' which type of error is occured during the test.

**Comment by The-Compiler** [comment]
Have you considered calling `pytest.main` instead ([How to invoke pytest — pytest documentation](https://docs.pytest.org/en/7.4.x/how-to/usage.html#calling-pytest-from-python-code)) and catching exceptions around that?
